### PR TITLE
fix loading repos from mac

### PR DESCRIPTION
**Pull request by hossamabdullah**
Fix loading of repos in ays

**Version**: 
9.3.0

### CODE REVIEW

File: JumpScale9AYS/ays/lib/AtYourServiceRepo.py [hunk 1]
```diff
@@ -61,17 +61,27 @@ def _searchAysRepos(self, path):
...
         for reponame in out.splitlines():
+            if j.sal.fs.isLink(reponame):
+                prefabInstance = j.tools.prefab.get()
+                osName = prefabInstance.platformtype.osname
+                if "darwin" in osName:
+                    cmd = "stat -f '%%Y' %s" % (reponame)
+                else:
+                    cmd = "readlink -f %s" % (reponame)
+                rc, out, err = j.sal.process.execute(cmd, die=False, showout=False)
+                reponame = out
```
Comment (ahussein) [inline JumpScale9AYS/ays/lib/AtYourServiceRepo.py]: I think the changes can be optimized a bit, I do not see the point of the loop, inside that new loop the code is doing:
- getting a prefab instance 
- creating new variable for the osName (should be renamed to os_name)
- checking if the osName match specific value 
the osname will not change, you only need to get it once.
I would prefer to check the os once and then construct the full command that is appropriate for the underlying os. No need for the loop.

File: JumpScale9AYS/ays/lib/AtYourServiceRepo.py [hunk 2]
```diff
@@ -61,17 +61,27 @@ def _searchAysRepos(self, path):
...
         res = []
-
-        cmd = """find %s \( -wholename '*.ays' \) -exec readlink -f {} \;""" % (path)
+        
+        cmd = "find %s \( -wholename '*.ays' \) " % (path)
         rc, out, err = j.sal.process.execute(cmd, die=False, showout=False)
+
         for reponame in out.splitlines():
+            if j.sal.fs.isLink(reponame):
+                prefabInstance = j.tools.prefab.get()
```
Comment (rkhamis) [inline JumpScale9AYS/ays/lib/AtYourServiceRepo.py]: just use j.tools.prefab.local.platformtype.osname

File: JumpScale9AYS/ays/lib/AtYourServiceRepo.py [hunk 3]
```diff
@@ -61,17 +61,27 @@ def _searchAysRepos(self, path):
...
+                osName = prefabInstance.platformtype.osname
+                if "darwin" in osName:
+                    cmd = "stat -f '%%Y' %s" % (reponame)
+                else:
+                    cmd = "readlink -f %s" % (reponame)
+                rc, out, err = j.sal.process.execute(cmd, die=False, showout=False)
```
Comment (rkhamis) [inline JumpScale9AYS/ays/lib/AtYourServiceRepo.py]: this doesn't make sense to me

I think maybe what you want to do is
```
cmd = "find %s \( -wholename '*.ays' \) " % (path)
os_name = j.tools.prefab.local.platformtype.osname
if "darwin" in os_name:
    cmd = "%s stat -f '%%Y' %s" % (cmd, reponame)
else:
    cmd = "%s readlink -f %s" % (cmd, reponame)
```

File: JumpScale9AYS/ays/lib/AtYourServiceRepo.py [hunk 4]
```diff
@@ -61,17 +61,27 @@ def _searchAysRepos(self, path):
...
+                reponame = out
             reponame = reponame.split(".ays")[0]
             if reponame.startswith(".") or reponame.startswith("_"):
                 continue
             res.append(reponame)
 
         return res
-
+
```
Comment (rkhamis) [inline JumpScale9AYS/ays/lib/AtYourServiceRepo.py]: please clean white spaces

File: JumpScale9AYS/ays/lib/AtYourServiceRepo.py [hunk 5]
```diff
@@ -61,17 +61,27 @@ def _searchAysRepos(self, path):
         path = j.sal.fs.pathNormalize(path)
 
         res = []
-
-        cmd = """find %s \( -wholename '*.ays' \) -exec readlink -f {} \;""" % (path)
+
```
Comment (rkhamis) [inline JumpScale9AYS/ays/lib/AtYourServiceRepo.py]: please clean white spaces

File: JumpScale9AYS/ays/lib/TemplateRepo.py [hunk 6]
```diff
@@ -29,12 +29,25 @@ def searchActorTemplates(path, is_global=False):
...
+                rc, out, err = j.sal.process.execute(cmd%templateName, die=False, showout=False)
+                templateName = out
+            res.add(templateName)
+        return res
+    else:
+        cmd = """find %s \( -wholename '*/templates/*actions.py' -or -wholename '*/templates/*schema.capnp' -or -wholename '*/templates/*config.yaml' -or -wholename '*/tests/*actions.py' -or -wholename '*/tests/*schema.capnp' -or -wholename '*/tests/*config.yaml' %s \) -exec readlink -f {} \;""" % (path, actortemplatessearch)
+        rc, out, err = j.sal.process.execute(cmd, die=False, showout=False)
+        if rc == 0:
+            return out.splitlines()
+        return res
```
Comment (ahussein) [inline JumpScale9AYS/ays/lib/TemplateRepo.py]: There are two problems with this:
- Multiple returns, this is normally not a very good practice. I would just make sure that we have one return statement if possible.
- Code duplication: we should be able to create a utility method and re-use it in both modules.

**Pull request merged**